src/Sulu/Bundle/SnippetBundle/DependencyInjection/Compiler/SnippetAreaCompilerPass.php

```diff
@@ -72,23 +75,36 @@ public function process(ContainerBuilder $container)
     }
 
     /**
-     * Get area.
+     * @param array{key: string, title: array<string>} $area
+     * @param array<string> $locales
+     * @param array<string, string> $templateTitles
      *
-     * @return array
+     * @return array{key: string, template: string, title: array<string, string>}
      */
-    private function getArea($template, $area, $locales, $templateTitles)
+    private function getArea(ContainerBuilder $container, string $template, array $area, array $locales, array $templateTitles): array
     {
         $key = $area['key'];
 
         $titles = [];
+        $areaTitles = $area['title'];
 
-        foreach ($locales as $locale) {
-            $title = $templateTitles[$locale] . ' ' . \ucfirst($key);
-            if (isset($area['title'][$locale])) {
-                $title = $area['title'][$locale];
+        // If we only have one title, it is probably a translation key
+        if (1 === \count($areaTitles)) {
+            $translator = $container->get('translator');
+            $titleToTranslate = \reset($areaTitles);
+            foreach ($locales as $locale) {
+                $titles[$locale] = $translator->trans($titleToTranslate, [], 'admin', $locale);
             }
+        } else {
+            \trigger_error('Translating snippet areas in the XML is deprecated use a translation instead.', \E_USER_DEPRECATED);
```

Comment:
this will not be deprecated by sulu. It is more common that translations for the admin are done in the xml not in translations files.

Reply:
Removed the deprecation message.